pit30m/data/log_reader.py

```diff
@@ -143,24 +148,50 @@ def __repr__(self) -> str:
         return f"Pit30M Log Reader: {self._log_root_uri}"
 
     @cached_property
-    def partitions_mask(self) -> np.ndarray:
+    def _partitions_mask(self) -> np.ndarray:
         """
         Returns a boolean np array that accounts for the requested partitions, by setting sensor readings that should
-        be skipped to False. Currently computed from the Front Camera.
+        be skipped to False. Computed for the raw pose data @ 100 Hz.
+
+        Returns:
+            A structured array with two fields:
+            1. "capture_time": float 64 series of pose GPS timestamps
+            2. "mask": boolean indicating whether each pose is included in the requested partitions
         """
-        if not self.partitions:
-            n_sensor_measurements = len(self.get_cam_geo_index(CamName.MIDDLE_FRONT_WIDE))
-            return np.full(n_sensor_measurements, True)
+        # breakpoint()
+        n_sensor_measurements = len(self.raw_pose_data)
+        partition_indices = self._partition_assigments
+        partition_indices += (np.full(n_sensor_measurements, True),)
+
+        combined_indices_mask = np.logical_and.reduce(partition_indices)
+        timestamps = self.raw_pose_data["capture_time"]
+
+        # HACK: Convert the raw poses' timestamps to UNIX to make them comparable to those... everywhere else
+        for i, timestamp in enumerate(timestamps):
+            if not np.isnan(timestamp):
+                timestamps[i] = gps_to_unix_timestamp(timestamp)
+        timestamps = rfn.unstructured_to_structured(timestamps[:, np.newaxis], dtype=[("timestamp", "<f8")])
+
+        combined_indices_mask = rfn.unstructured_to_structured(
+            combined_indices_mask[:, np.newaxis], dtype=[("mask", "?")]
+        )
 
-        partition_indices = self.partition_assigments
-        combined_indices = np.logical_and.reduce(partition_indices)
-        return combined_indices
+        mask_with_timestamps = rfn.merge_arrays((timestamps, combined_indices_mask))
+        return mask_with_timestamps
 
     @property
     @memory.cache(verbose=0)
-    def partition_assigments(self):
-        """Fetches partition indices from S3 and converts them to boolean arrays according to the reader partition values"""
+    def _partition_assigments(self) -> Tuple[np.ndarray]:
+        """Fetches partition indices from S3 and converts them to boolean arrays according to the reader's partition
+        values. Note that these partitions are done wrt the raw pose data @ 100Hz, not the sensor data; therefore,
+        we must index sensor data into these partitions to find out which partitions they belong to.
+
+        Returns:
+            A tuple of boolean arrays. Each array has n_dense_poses entries, and each entry represents membership of the
```

Comment:
How many elements? If the count will vary based on the context, would a `list` be a better fit?

Reply:
Right, moved it to a list!